js/upload-form.js

```diff
@@ -0,0 +1,51 @@
+import './validation.js';
+import './filters-slider.js';
+import {addEventListenerRest, isEscapeKeydown, removeEventListenerRest, stopPropagationFunc} from './functions.js';
+
+const ESC_RESISTANT_CLASS = ['input[name="hashtags"]', 'textarea[name="description"]'];
+
+const uploadButton = document.querySelector('#upload-file');
+const uploadModal = document.querySelector('.img-upload__overlay');
+const uploadModalCloseButton = uploadModal.querySelector('#upload-cancel');
+const uploadImgPreview = uploadModal.querySelector('.img-upload__preview').children[0];
+
+const createUploadForm = () => {
+  //#region
+  const showModal = () => {
+    uploadImgPreview.src = `photos/${uploadButton.value.split('\\')[2]}`;
+    uploadModal.classList.remove('hidden');
+    document.body.classList.add('modal-open');
+    addEventListenerRest(uploadModal, 'keydown', stopPropagationFunc, ...ESC_RESISTANT_CLASS);
+  };
+
+  const hideModal = () => {
+    uploadButton.value = '';
+    uploadModal.classList.add('hidden');
+    uploadModal.querySelector('.pristine-error').style.display = 'none';
```

Comment:
Если ошибок при валидации формы не было, то мы не найдем элемент `uploadModal.querySelector('.pristine-error') // вернет null`

```suggestion
    const pristineError = uploadModal.querySelector('.pristine-error');
    
    if (pristineError !== null) {
        pristineError.style.display = 'none';
    }
```

https://learn.javascript.ru/optional-chaining